service_worker/index.ts

```diff
@@ -25,23 +26,10 @@ const mainImageName = 'mainImage';
 
 declare var self: ServiceWorkerGlobalScope;
 
-self.addEventListener('install', function (event): void {
-  // precache all assets
-  event.waitUntil(
-    caches.open(cacheName).then(function (cache) {
-      return cache.addAll([
-        '__APP_ROOT__/',
-        '__APP_ROOT__/index.html',
-        '__APP_ROOT__/index.ts',
-        '__APP_ROOT__/manifest.json',
-        '__APP_ROOT__/sw.ts',
-      ]);
-    }),
-  );
-});
+precacheAndRoute(self.__WB_MANIFEST);
```

Comment:
with this here, does the StaleWhileRevalidate strategy ever actually get used? (and if not, then how does the cache get refreshed when new content is pushed to the server?)

Reply:
I'd be keen to figure out how to test service worker behaviour, it's something I frequently struggle with in my own projects.

That said, to your questions, for the files listed in `__WB_MANIFEST` we're now applying a cache-first strategy. That's definitely the right thing to do now that our JavaScript bundle has a hash. Other, non-generated content (i.e. data in `/public/`) will still be served according to SWR.